cobc/replace.c

```diff
@@ -0,0 +1,841 @@
+/*
+   Copyright (C) 2001-2023 Free Software Foundation, Inc.
+
+   Authors:
+   Keisuke Nishida, Roger While, Ron Norman, Simon Sobisch, Brian Tiffin,
+   Edward Hart, Dave Pitts, Fabrice Le Fessant
+
+   This file is part of GnuCOBOL.
+
+   The GnuCOBOL compiler is free software: you can redistribute it
+   and/or modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of the
+   License, or (at your option) any later version.
+
+   GnuCOBOL is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with GnuCOBOL.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "tarstamp.h"
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdarg.h>
+#include <string.h>
+#ifdef	HAVE_STRINGS_H
+#include <strings.h>
+#endif
+#include <ctype.h>
+#include <time.h>
+#include <limits.h>
+
+#include "cobc.h"
+#include "tree.h"
+#include "replace.h"
+
+#define TRUE  1
```

Comment:
Please don't use TRUE/FALSE pseudo-boolean in 3.x; just go with 0 and 1 in the code for now and in most cases instead of storing the result to 0/1 you can directly `return 1`.

Reply:
Why not ? I am not sure to understand why these (local) macros are a problem in 3.x

Reply:
(anyway, I removed them in the last commit)

Reply:
Only because this should be consistent between these files and 4.x should use boolean types ("backported" via gnulib for old compilers).